File: front_end/src/types/GoodsReceipt.ts
export const GoodsReceiptType = {
    Purchase: 1,
    Return: 2,
    Transfer: 3
} as const;
export type GoodsReceiptType = typeof GoodsReceiptType[keyof typeof GoodsReceiptType];

export const GoodsReceiptStatus = {
    Draft: 0,
    Completed: 1,
    Cancelled: 2
} as const;
export type GoodsReceiptStatus = typeof GoodsReceiptStatus[keyof typeof GoodsReceiptStatus];

export interface GoodsReceiptSummaryDto {
    id: string;
    code: string;
    type: GoodsReceiptType;
    referenceCode: string;
    receiptDate: string;
    status: GoodsReceiptStatus;
    totalQuantity: number;
    createdDate: string;
}

export interface GoodsReceiptLineDto {
    id: string;
    productId: string;
    productName: string;
    uoMId: string;
    uoMName: string;
    conversionRate: number;
    expectedQuantity: number;
    actualQuantity: number;
    notes: string;
}

export interface GoodsReceiptDto {
    id: string;
    code: string;
    type: GoodsReceiptType;
    referenceId?: string | null;
    referenceCode: string;
    supplierId?: string | null;
    warehouseId: string;
    receiptDate: string;
    notes: string;
    status: GoodsReceiptStatus;
    totalQuantity: number;
    createdDate: string;
    lines: GoodsReceiptLineDto[];
}

export interface CreateGoodsReceiptLineDto {
    productId: string;
    uoMId: string;
    conversionRate: number;
    expectedQuantity: number;
    actualQuantity: number;
    notes: string;
}

export interface CreateGoodsReceiptDto {
    type: GoodsReceiptType;
    referenceId?: string | null;
    referenceCode: string;
    supplierId?: string | null;
    warehouseId: string;
    receiptDate: string;
    notes: string;
    lines: CreateGoodsReceiptLineDto[];
}

export interface UpdateGoodsReceiptLineDto {
    id?: string | null;
    productId: string;
    uoMId: string;
    conversionRate: number;
    expectedQuantity: number;
    actualQuantity: number;
    notes: string;
}

export interface UpdateGoodsReceiptDto {
    referenceId?: string | null;
    referenceCode: string;
    warehouseId: string;
    receiptDate: string;
    notes: string;
    lines: UpdateGoodsReceiptLineDto[];
}

export interface ResolveBarcodeDto {
    productId: string;
    productName: string;
    uoMId: string;
    uoMName: string;
    conversionRate: number;
    isBaseUnit: boolean;
}

File: back_end/Sales.Domain/Entities/Inventory/GoodsReceipt.cs
using System;
using System.Collections.Generic;
using Sales.Domain.Enums;
using Sales.Domain.Interfaces;

namespace Sales.Domain.Entities.Inventory
{
    /// <summary>
    /// Thực thể đại diện cho một Phiếu Nhập Kho.
    /// </summary>
    public class GoodsReceipt : ISoftDelete
    {
        /// <summary>Khóa chính (UUID).</summary>
        public Guid Id { get; set; }

        /// <summary>Mã phiếu nhập kho (Ví dụ: GR-YYMM-00001).</summary>
        public string Code { get; set; } = string.Empty;

        /// <summary>Loại phiếu nhập kho.</summary>
        public GoodsReceiptType Type { get; set; }

        /// <summary>Khóa ngoại trỏ đến chứng từ gốc (Purchase Order, Return Order, Transfer Order).</summary>
        public Guid? ReferenceId { get; set; }

        /// <summary>Mã chứng từ gốc để hiển thị nhanh.</summary>
        public string ReferenceCode { get; set; } = string.Empty;

        /// <summary>Khóa ngoại trỏ đến Nhà cung cấp (Lấy từ PO). FK -> Customers (hoặc Suppliers).</summary>
        public Guid? SupplierId { get; set; }

        /// <summary>Kho nhập. FK -> Warehouses (nếu có, ở đây có thể lưu dưới dạng tham chiếu hoặc enum nếu không có bảng Warehouse).</summary>
        public Guid WarehouseId { get; set; }

        /// <summary>Ngày nhập kho thực tế.</summary>
        public DateTime ReceiptDate { get; set; }

        /// <summary>Ghi chú cho phiếu nhập kho.</summary>
        public string Notes { get; set; } = string.Empty;

        /// <summary>Trạng thái của phiếu nhập kho.</summary>
        public GoodsReceiptStatus Status { get; set; } = GoodsReceiptStatus.Draft;

        /// <summary>Tổng số lượng thực tế của tất cả các dòng sản phẩm.</summary>
        public decimal TotalQuantity { get; set; }

        // --- ISoftDelete ---
        /// <summary>Cờ xóa mềm: true = đã bị xóa mềm, không hiển thị trong các query thông thường.</summary>
        public bool IsDeleted { get; set; } = false;

        /// <summary>Ngày giờ thực hiện xóa mềm (UTC). Null nếu chưa xóa.</summary>
        public DateTime? DeletedDate { get; set; }

        /// <summary>ID người dùng thực hiện xóa mềm.</summary>
        public Guid? DeletedBy { get; set; }

        // --- Audit ---
        /// <summary>Ngày giờ tạo bản ghi (UTC).</summary>
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

        /// <summary>ID người dùng tạo bản ghi.</summary>
        public Guid? CreatedBy { get; set; }

        /// <summary>Ngày giờ cập nhật gần nhất (UTC). Null nếu chưa cập nhật.</summary>
        public DateTime? UpdatedDate { get; set; }

        /// <summary>ID người dùng cập nhật lần cuối.</summary>
        public Guid? UpdatedBy { get; set; }

        // --- Navigation Properties ---
        /// <summary>Danh sách các dòng sản phẩm chi tiết trong phiếu nhập.</summary>
        public virtual ICollection<GoodsReceiptLine> Lines { get; set; } = new List<GoodsReceiptLine>();
    }
}
